zerver/openapi/zulip.yaml

```diff
@@ -6666,18 +6690,24 @@ paths:
                         description: |
                           The same `anchor` specified in the request (or the computed one, if
                           `use_first_unread_anchor` is `true`).
+
+                          Only present if `message_ids` is not provided.
                       found_newest:
                         type: boolean
                         description: |
                           Whether the server promises that the `messages` list includes the very
                           newest messages matching the narrow (used by clients that paginate their
                           requests to decide whether there may be more messages to fetch).
+
+                          Only present if `message_ids` is not provided.
```

Comment:
In the `#api design` thread, the proposal was for `found_newest` and `found_oldest` to continue to be always present:
https://chat.zulip.org/#narrow/stream/378-api-design/topic/Message.20matches.20narrow/near/1951604